tests/query.json

```diff
@@ -201,6 +203,7 @@
         ["QueryCount", 5, "StringObject", "stringCol BEGINSWITH[c] 'A'"],
         ["QueryCount", 4, "StringObject", "stringCol ENDSWITH[c] 'c'"],
         ["QueryCount", 2, "StringObject", "stringCol CONTAINS[c] 'B'"]
+        ["QueryCount", 2, "StringObject", "stringCol LIKE[c] '*B?'"]
```

Comment:
It'd be good to test handling of invalid predicates like `stringCol LIKE 1`, and substituted patterns like `stringCol LIKE $0`